web/templates/stream_settings/stream_settings.hbs

```diff
@@ -28,13 +28,19 @@
           id="preview-stream-button"
           hidden=(not should_display_preview_button)
           }}
-        {{#if can_archive_stream}}
+        {{#if (or is_realm_admin is_creator)}}
```

Comment:
This is not correct as all users with permission to administer channel have permission to archive or unarchive a channel and this is not the correct way to check whether user can administer channel, it will depend on `can_administer_channel_group`.

Would be good to let `update_settings_button_for_archive_and_unarchive` function handle this and have these buttons present in DOM always, so that we can also live-update the UI correctly when administering permission changes in future.